src/OpalCompiler-Core/OpalCompiler.class.st

```diff
@@ -221,13 +224,16 @@ OpalCompiler >> ast: anObject [
 
 { #category : #accessing }
 OpalCompiler >> bindings [
-	^ self compilationContext bindings
+	^ bindings
 ]
 
 { #category : #accessing }
-OpalCompiler >> bindings: aDictionary [
+OpalCompiler >> bindings: aCollectionOfAssociations [
 	"allows to define additional binding, note: Globals are not shadowed"
-	self compilationContext bindings: aDictionary
+	"bindings are LiteralVariables, not simple Associations.
+	This method receives any collection with Associations or Variables inside and converts those to AdditionalBindings"
+	bindings := (aCollectionOfAssociations asDictionary associations
```

Comment:
The `asDictionary associations` seems redondant if we already have a collection of associations. Is it to cast LiteralVariables into Associations? Then why do the #isAssociation check?
I wonder if there is a real need for this or if this is just a lost on perfs.

Of course this is not a proble of the current PR since it was already done like this.

Reply:
I just moved the method. But you're right, that weird. Maybe @MarcusDenker knows if there is a real need or if it is some leftover code